wallet/src/de/schildbach/wallet/ui/HeaderBalanceFragment.java

```diff
@@ -184,12 +192,28 @@ public void onEvent(SyncProgressEvent event) {
     }
 
     private void updateView() {
+        View balances = view.findViewById(R.id.balances);
+        TextView walletBalanceSyncMessage = view.findViewById(R.id.wallet_balance_sync_message);
+        View balancesLayout = view.findViewById(R.id.balances_layout);
+
+        if (hideBalance) {
+            balancesLayout.setVisibility(View.GONE);
+            caption.setText(R.string.home_balance_hidden);
+            hideShowBalanceHint.setText(R.string.home_balance_show_hint);
+            balances.setVisibility(View.GONE);
+            walletBalanceSyncMessage.setVisibility(View.GONE);
+            showBalanceButton.setVisibility(View.VISIBLE);
+            return;
+        }
+        balancesLayout.setVisibility(View.VISIBLE);
+        caption.setText(R.string.home_available_balance);
+        hideShowBalanceHint.setText(R.string.home_balance_hide_hint);
+        showBalanceButton.setVisibility(View.GONE);
+
         if (!isAdded()) {
             return;
         }
 
-        View balances = viewBalance.findViewById(R.id.balances);
-        TextView walletBalanceSyncMessage = viewBalance.findViewById(R.id.wallet_balance_sync_message);
         if (!isSynced) {
```

Comment:
If the app is not synced, should we also hide "tap to hide balance"?

Reply:
We shouldn't because user might want to hide balance before it shows up